app/models/competition.rb

```diff
@@ -379,6 +379,13 @@ def advancement_condition_must_be_present_for_all_non_final_rounds
     errors.add(:registration_close, I18n.t('competitions.errors.registration_already_closed')) if !editing_user.can_admin_competitions? && registration_range_specified? && registration_past?
   end
 
+  validate :payments_nil_to_change_currency
+  private def payments_nil_to_change_currency
+    return unless currency_code_changed?
```

Comment:
This can directly become an `if: :currency_code_changed?` option in the `validate` declaration

Reply:
Changed